test/test_torch.py

```diff
@@ -153,6 +155,53 @@ def make_contiguous_slice(size, dtype):
 
         return tensors
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    @skipIfRocm
+    def test_cublas_multiple_threads_same_device(self):
+        size = 16
+        x = torch.ones((size, size), device='cuda')
+        weight = torch.ones((size, size), device='cuda')
+
+        results = {}
+
+        num_threads = 100
+        trials = 10
+        test_iters = 100
+        barrier = threading.Barrier(num_threads)
+
+        def _worker(t, input_):
+            my_stream = torch.cuda.Stream()
+            results[t] = input_
+            with torch.cuda.stream(my_stream):
+                barrier.wait()
```

Comment:
It's cool that the barrier makes the test more consistent!  Can we also add a barrier to the [cudnn test](https://github.com/zasdfgbnm/pytorch/blob/cublas-handle/test/test_nn.py#L3132)?  I never had problems with the cudnn test's consistency before but if a barrier is beneficial for the cublas version we should include it in both places.